### PR TITLE
Allow `full_path` in addition to `id` in gitlab_project data source

### LINKED ISSUE #140: [Feature Request] use full_path to define gitlab_project data source
Hi!

While refactoring my terraform repository I have found out that it would be awesome to provide same way to create gitlab_project data source as for gitlab_group by using full_path instead of id, this woulde make a code more human readable :).

### PR DESCRIPTION
**Pull request by sirlatrom**
Fixes #140.

### CODE REVIEW

**Review by armsnyder — CHANGES_REQUESTED** [review]
Thanks for the contribution. A better fix would be to change the type for `id` into `string` in the resource schema (this is a backward compatible change) and update the documentation to mention that id can be an int or string. I don't think `full_path` is the standard name for this field in the GitLab API, nor is it used anywhere else in this plugin.

**Comment by sirlatrom** [comment]
> 
> 
> Thanks for the contribution. A better fix would be to change the type for `id` into `string` in the resource schema (this is a backward compatible change) and update the documentation to mention that id can be an int or string. I don't think `full_path` is the standard name for this field in the GitLab API, nor is it used anywhere else in this plugin.

It's used here: https://github.com/gitlabhq/terraform-provider-gitlab/blob/f821d591b6fdb7e388c3eed2b637e758900fb553/gitlab/data_source_gitlab_group.go#L23-L30 which is what I took as inspiration.

**Comment by armsnyder** [comment]
> It's used here:
> 
> https://github.com/gitlabhq/terraform-provider-gitlab/blob/f821d591b6fdb7e388c3eed2b637e758900fb553/gitlab/data_source_gitlab_group.go#L23-L30
> 
> which is what I took as inspiration.

Ahh, good point. However, I would still rather stick to the official GitLab API field names, as I see this as API debt.

See the API fields names for GET Project: https://docs.gitlab.com/ee/api/projects.html#get-single-project

See the HashiCorp guidance on this subject: https://www.terraform.io/docs/extend/hashicorp-provider-design-principles.html#resource-and-attribute-schema-should-closely-match-the-underlying-api

**Comment by sirlatrom** [comment]
> Ahh, good point. However, I would still rather stick to the official GitLab API field names, as I see this as API debt.
> 
> See the API fields names for GET Project: https://docs.gitlab.com/ee/api/projects.html#get-single-project
> 
> See the HashiCorp guidance on this subject: https://www.terraform.io/docs/extend/hashicorp-provider-design-principles.html#resource-and-attribute-schema-should-closely-match-the-underlying-api

OK, I'll change the type of `id`, but are you quite sure you can go from a `TypeInt` to `TypeString` without state parsing issues? I'll give it a try.

**Comment by armsnyder** [comment]
> OK, I'll change the type of `id`, but are you quite sure you can go from a `TypeInt` to `TypeString` without state parsing issues? I'll give it a try.

Yep, going from int -> string is fine. Terraform will update the type in the state on the next apply, and users can even continue to set this field to a number in their config.

**Comment by sirlatrom** [comment]
> Yep, going from int -> string is fine. Terraform will update the type in the state on the next apply, and users can even continue to set this field to a number in their config.

@armsnyder Done in 55a5f54.

File: docs/data-sources/project.md [hunk 1]
```diff
@@ -10,11 +10,19 @@ data "gitlab_project" "example" {
...
+```
+
 ## Argument Reference
 
 The following arguments are supported:
 
-* `id` - (Required) The integer that uniquely identifies the project within the gitlab install.
+* `id` - (Required) The integer or path with namespace that uniquely identifies the project within the gitlab install.
+
+**Note**: Exactly one of `id` or `full_path` must be provided.
```
Comment (armsnyder) [inline docs/data-sources/project.md:25]: Can you revert this note please

**Review by armsnyder — CHANGES_REQUESTED** [review]
Great! One final note.

File: docs/data-sources/project.md (hunk 1 above)
Comment (sirlatrom) [reply docs/data-sources/project.md]: Yup, sorry I missed that. 091c23f should do it.

**Review by armsnyder — APPROVED** [review]
Thanks!

**Pull request merged**